Remove Mat/Vec template parameters #2

norihiro-w opened the pull request:
follow up of #1257

chleh commented on ProcessLib/TES/TESProcess.cpp:
```diff
@@ -9,6 +9,8 @@
 
 #include "TESProcess.h"
 
+#include "Utils/CreateLocalAssemblers.h"
```
Sall we use a full path here, i.e. prepend `ProcessLib`?

chleh commented on NumLib/ODESolver/MatrixTranslator.h:
```diff
@@ -25,85 +25,80 @@ namespace NumLib
 
 /*! Translates matrices and vectors assembled by a provided ODE (or other equation)
- *  to the stiffness matrix and right-hand side vector of a linear equation system
+ *  to the stiffness GlobalMatrix and right-hand side vector of a linear equation system
```
I'd keep the old comment.

chleh commented on MathLib/LinAlg/LinearSolver.h:
```diff
@@ -20,50 +20,15 @@ class ConfigTree;
 
-//! Basic linear solver interface.
-template<typename Matrix, typename Vector>
-class LinearSolver
```
If you kept the linear solver base class, we could in principle have EigenLinearSolver and EigenLISLinearSolver (and later on maybe others, too) in a single build.

norihiro-w replied:
we can discuss it when such a feature is required

chleh commented:
Nice :+1:. Maybe now even the compile time reduces a bit?
Now some lines in NumLib/ODESolver are even longer than before. Could you clang-format everything in NumLib/ODESolver, please? I hope that won't change things (i.e. doc comments) to the worse.

endJunction commented:
There is a clang-format-diff python script for application of clang-format to the diffs only.

norihiro-w commented:
formatted

endJunction commented on NumLib/DOF/GlobalMatrixProviders.h:
```diff
@@ -14,17 +14,14 @@
 struct GlobalVectorProvider
 {
-    static VectorProvider<Vector>& provider;
+    static VectorProvider<GlobalVector>& provider;
```
Not following all the ideas here, just a question: Why is GlobalVector template argument needed? Couldn't the VectorProvider just use the global type of vectors?

chleh replied:
Should be possible since the Eigen::MatrixXd cases are removed from TestODEInt.cpp

norihiro-w replied:
This will be done in #1283

endJunction commented on NumLib/ODESolver/MatrixTranslator.h:
```diff
@@ -12,374 +12,249 @@
-template<typename Matrix, typename Vector>
-class MatrixTranslator<Matrix, Vector, ODESystemTag::FirstOrderImplicitQuasilinear>
+template <>
+class MatrixTranslator<ODESystemTag::FirstOrderImplicitQuasilinear>
```
I think the specializations can also go into the cpp file, but I might be wrong about this.

chleh replied:
No, the specializations are entirely different types. So the declarations of the members must stay here.

norihiro-w replied:
This will be done in #1283

endJunction commented:
:fast_forward: so far.
I'd have split this PR in several parts, because reviewing +-2kloc changes is difficult and error prone. I definitely didn't get all the changes and some changes are "mechanical" (like sed scripts) other changes are by hand and need extra attention (which they didn't get from my side).

norihiro-w commented:
@endJunction Have you made review commit by commit? Basically I deleted the template parameters and moved some implementations to cpp files. Large diffs are caused by clang-format.

endJunction commented:
In single chunk.

norihiro-w commented:
i will try restructuring commits for helping reviews. let me know if you have suggestions

norihiro-w commented:
changed commits to include removing the template parameters in NumLib/ODESolver dir and resulting changes in other classes. see commits titled "rm Mat/Vec templates from xxxx".

norihiro-w commented:
Jenkins, test this please!

norihiro-w commented:
build time: `make -j4` for cli, test, dynamic link, no lis
2m52.551s (before this PR) vs 2m50.176s (with this PR)

endJunction commented:
@norihiro-w Would you please rebase it?

norihiro-w commented:
rebased

endJunction commented on NumLib/ODESolver/EquationSystem.h:
```diff
@@ -30,9 +29,7 @@ enum class IterationResult : char
  *
  * \tparam Vector the type of the solution vector of the equation.
```
I didn't see it before, but all of the Doxygen documentation needs a fixup; the template parameter does not exist anymore.

norihiro-w replied:
i will fix them
